src/massive/munit/TestRunner.hx

```diff
@@ -182,29 +189,82 @@ class TestRunner implements IAsyncDelegateObserver
         suiteIndex = 0;
         clientCompleteCount = 0;
         Assert.assertionCount = 0; // don't really like this static but can't see way around it atm. ms 17/12/10
+        asyncDelegates = new Array<AsyncDelegate>();
         testSuites = [for(suiteType in testSuiteClasses) Type.createInstance(suiteType, emptyParams)];
         startTime = Timer.stamp();
 
-        #if (neko || cpp || java)
-		var self = this;
-		var runThread:Thread = Thread.create(function()
-		{
-			self.execute();
-			while (self.running)
-			{
-				Sys.sleep(.2);
-			}
-			var mainThead:Thread = Thread.readMessage(true);
-			mainThead.sendMessage("done");
-		});
-		runThread.sendMessage(Thread.current());
-		Thread.readMessage(true);
+        #if (!lime && !nme && (neko || cpp || java))
```

Comment:
Sorry I didn't see these comments!

I believe this code is to prevent applications which have no main loop from exiting immediately.

I agree it would be nice if this was a little more generic, but there's no certain way to say that you're using a framework that is aware of munit and can provide a main loop. I guess an munit-specific define that needs to be added would be a fair compromise